controllers/configurationpolicy_controller.go

```diff
@@ -1709,6 +1710,29 @@ func (r *ConfigurationPolicyReconciler) determineDesiredObjects(
 		return nil, &scopedGVR, event, nil
 	}
 
+	// For mustnothave with no name and with an object selector, filter the "desired" objects
+	// so that only ones matching the other details in the template are included.
+	if objectT.ComplianceType.IsMustNotHave() && parsedMinMetadata.Metadata.Name == "" && objectSelector != nil {
+		targetedObjects := make([]*unstructured.Unstructured, 0, len(desiredObjects))
+
+		for _, d := range desiredObjects {
+			unnamedObj := d.DeepCopy()
+			unnamedObj.SetName("")
+
+			matchingNames, _ := r.getMatchingNames(plc, unnamedObj, scopedGVR, objectT)
+
+			for _, n := range matchingNames {
+				if n == d.GetName() {
```

Comment:
(Optional) Minor optimization: `d.GetName()` could be declared as a variable outside this loop. (Though it'd also make it busier, so maybe it's better leave it as-is. 🙂 )